api/v1alpha1/bucketcontent_types.go

```diff
@@ -17,25 +17,96 @@ limitations under the License.
 package v1alpha1
 
 import (
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+// Regions where AWS S3 API is available
+type S3Region string
+
+const (
+	// S3RegionUSEast1 = us-east-1
+	S3RegionUSEast1 S3Region = `json:"us-east-1"`
+)
+
+// S3ProtocolSpec is the specification for S3 and S3 compatible Object Storage services
+type S3ProtocolSpec struct {
+	// Endpoint is the URL for accessing the S3 endpoint
+	// +optional
+	Endpoint string `json:"endpoint,omitempty"`
+
+	// BucketName is the name of the S3 or S3 compatible bucket
+	BucketName string `json:"bucket"`
+
+	// Region is the Region where the bucket is situated
+	// +optional
+	Region S3Region `json:"region,omitempty"`
+
+	// SignatureVersion is the version of the S3 API signature
+	// +optional
+	SignatureVersion string `json:"signatureVersion,omitempty"`
+
+	// ServiceAccount is the service account for the BucketContent
+	// +optional
+	ServiceAccount string `json:"serviceAccount,omitempty"`
```

Comment:
`ServiceAccount` is actually part of the BucketAccess side of the API - I'm not sure storage APIs should have any references to user identities.